### PR TITLE
AO3-4963 Modify new taggings_count code to handle brand new fandoms used on External Works.

### PR DESCRIPTION
**Pull request by tickinginstant**
## Issue

https://otwarchive.atlassian.net/browse/AO3-4963

## Purpose

This pull request moves the call to `Tag.update_tag_cache` from the Work's `expire_caches` function into a new callback in the Tagging class, which is triggered whenever a Tagging is created or destroyed. This ensures that the `taggings_count` will behave the same regardless of whether the new tag is used on a Work or an ExternalWork (or a Bookmark).

## Testing

There are steps for replicating the bug in the JIRA issue.

## Credit

tickinginstant, she/her

### CODE REVIEW

File: features/tags_and_wrangling/brand_new_fandoms.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    Given basic tags
+      And the tag "My Brand New Fandom" does not exist
+
+  Scenario: Brand new fandoms should be visible on the uncategorized fandoms page.
+    Given I am logged in as a random user
+      And I post a work "My New Work" with fandom "My Brand New Fandom"
+      And The periodic tag count task is run
+    When I follow "Uncategorized Fandoms" within "#header"
+    Then I should see "My Brand New Fandom"
+
```
Comment (zz9pzza) [inline features/tags_and_wrangling/brand_new_fandoms.feature]: Could we have a follow on test that checks that the count is zero if the last external work to use a tag is destroyed ?

Comment (tickinginstant) [inline features/tags_and_wrangling/brand_new_fandoms.feature]: I added some tests checking whether the fandom disappears from the Uncategorized Fandoms/Fandoms by Media page when it should. Do you want me to try to add in some RSpec tests that explicitly check whether the count is zero, too?

Comment (zz9pzza) [inline features/tags_and_wrangling/brand_new_fandoms.feature]: I think that the tests you have added are sufficient :)

If you feel like gilding the lily then do but don't feel obliged.

Comment (tickinginstant) [inline features/tags_and_wrangling/brand_new_fandoms.feature]: I think I'll leave it as is. :)
